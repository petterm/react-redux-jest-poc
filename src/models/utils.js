export default function makeActionCreator(type, ...argNames) {
  return (...args) => {
    const action = { type };
    if (argNames.length > 0) {
      action.payload = {};
      argNames.forEach((arg, index) => {
        action.payload[argNames[index]] = args[index];
      });
    } else if (typeof args[0] !== 'undefined') {
      action.payload = args[0];
    }
    return action;
  };
}
