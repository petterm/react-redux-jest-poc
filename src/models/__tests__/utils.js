import { makeActionCreator } from '../utils';

describe('config utils', () => {
  it('can make action-creators', () => {
    expect(makeActionCreator('SOME_ACTION_TYPE')()).toMatchSnapshot();
    expect(makeActionCreator('SOME_ACTION_TYPE')('value-1')).toMatchSnapshot();
    expect(makeActionCreator('SOME_ACTION_TYPE', 'arg-1', 'arg-2')('value-1', 'value-2'))
      .toMatchSnapshot();
  });
});
