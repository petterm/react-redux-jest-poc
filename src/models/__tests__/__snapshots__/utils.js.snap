exports[`config utils can make action-creators 1`] = `
Object {
  "type": "SOME_ACTION_TYPE",
}
`;

exports[`config utils can make action-creators 2`] = `
Object {
  "payload": "value-1",
  "type": "SOME_ACTION_TYPE",
}
`;

exports[`config utils can make action-creators 3`] = `
Object {
  "payload": Object {
    "arg-1": "value-1",
    "arg-2": "value-2",
  },
  "type": "SOME_ACTION_TYPE",
}
`;
